docs.md

```diff
@@ -1,23 +1,499 @@
+GITHUB_USERNAME=<your_github_username>
+GITHUB_URL=<your_github_url where the authorization code lives>
+CLIENT_ID=<your_github_client_id>
+CLIENT_SECRET=<your_github_client_secret>
+SECRET_KEY=<your_secret_key>
+ALGORITHM=<your_algorithm e.g HS256>
+ENVIRONMENT=<your_environment e.g. production|development>
+```
+
+- Run the application from the entry point
+
+```console
+$ python3 main.py
+```
+- Open [localhost:8000/docs](localhost:8000/docs) for API Documentation
+
+### - Build the application image and start the container -
+   ```console
+   $ Configure environment variables - the .env file
+   $ docker-compose -f docker-compose.dev.yml up -d
+
+   ```
+### - Pull the official image from Docker Hub and start the container 
+
+   ```console
+   $ docker pull rscheele3214/rego-builder:latest
+   $ docker run rscheele3214/rego-builder:latest
+   ```
+
+
+Translation from JSON to REGO.
+============
+The REGO policies are created by a set of functions called **commands** that serves as the translation logic. Each of these commands is responsible for writing a specific REGO rule to a `.rego` file. However, the JSON to REGO conversion must follow certain conditions as defined by a pydantic model for effectiveness. 
+The following is how a rule is defined. The  Policy object which is the request body, as well as the Rule object is pydantic models.
+
+```py3
+class Policy:
+    name: str
+    rules: List[List[Rule]]
+```
+Rule Object: <br />
+```py3
+class Rule(BaseModel):
+    command: str
+    properties: Dict[str, Union[str, List[str], Dict[str, Union[str, List[str]]]]]
+```
+
+A rule is defined by two keys: command and properties. The command key holds one of the recognized commands and the properties key, holds another dictionary containing the input to the command function e.g `input_property` and `value`. in special cases, the `datasource_item` items are also included in the properties key.
+
+   ```json     
+   {
+      "command": "input_prop_equals",
+      "properties": {
+         "input_property": "request_method",
+         "value": "GET"
+      }
+   }
+   ```
+In the example above, the command key represents the operation to be performed and the properties key represents the properties that are being used in the operation.
+
+`input_prop_equals` is the command in the example that initiates the appropriate operation, on the properties object.
+The above rule translates to an equality check between the input property `input_property` and the value `value`. <br /> <br />
+The REGO equivalent of the above rule object is: <br />
+```rego
+   `input.request_method == "GET"`
+```
+
+Each Rule object forms a specific rule in a Allow block, and a list of Rules forms a Allow block. <br />
+```py3
+List[Rule] == Allow {
+   ...
+}
+```
+
+The API supports the following commands; input_prop_equals, input_prop_in, input_prop_in_as, allow_full_access.
+
+Usage
+============
+### Detailed explanation of commands with examples: 
```

Comment:
I think we can add a sentence about what this section entails instead of presenting another heading.